04.ls/ls.rb

```diff
@@ -57,4 +79,92 @@ def puts_table(entry_name_table, max_width)
   end
 end
 
+def create_entry_info_table
+  entry_info_table = Dir.glob('*').map do |entry_name|
+    entry_info = {}
+    stat = File::Stat.new(entry_name)
+    stat_mode = stat.mode.to_s(8)
+    entry_info[:file_mode] = get_file_mode(stat_mode, entry_name)
+    entry_info[:nlink] = stat.nlink.to_s
+    entry_info[:owner_name] = get_owner_name(stat)
+    entry_info[:group_name] = get_group_name(stat)
+    entry_info[:file_size] = stat.size.to_s
+    updated_day = File.mtime(entry_name)
+    entry_info[:updated_dates] = get_updated_dates(updated_day)
+    entry_info[:entry_name] = entry_name
+    entry_info
+  end
+  align_width(entry_info_table)
+end
+
+def create_total_blocks
+  Dir.glob('*').sum do |entry_name|
+    File::Stat.new(entry_name).blocks
+  end
+end
+
+def get_file_mode(stat_mode, entry_name)
+  aligned_stat_mode = stat_mode.to_s.length == 5 ? stat_mode.rjust(6, '0') : stat_mode
+  file_mode = []
+  file_mode << get_file_type(aligned_stat_mode)
+  file_mode << get_file_permission(aligned_stat_mode)
+  file_mode << get_extended_attributes(entry_name)
+  file_mode.join
+end
+
+def get_file_type(stat_mode)
+  FILE_TYPES[stat_mode[0, 2]]
+end
+
+def get_file_permission(stat_mode)
+  stat_mode[3..5].chars.map { |digit| FILE_PERMISSION_TABLES[digit] }.join
+end
+
+def get_extended_attributes(file_name)
+  attrs = `xattr #{file_name}`
+  attrs.empty? ? ' ' : '@'
+end
+
+def get_owner_name(stat)
+  Etc.getpwuid(stat.uid).name
+end
+
+def get_group_name(stat)
+  Etc.getgrgid(stat.gid).name
+end
+
+def get_updated_dates(updated_day)
+  six_month_ago = (Date.today << SIX_MONTH).to_time
+  six_month_ago_flag = six_month_ago >= updated_day
+  updated_date_format = six_month_ago_flag ? '%_m %e %_5Y' : '%_m %e %H:%M'
+  updated_day.strftime(updated_date_format)
+end
+
+def put_entry_names_infos(total_blocks, entry_info_table)
+  puts "total #{total_blocks}"
+  entry_info_table.map do |entry_info|
+    puts entry_info.join(' ')
+  end
+end
+
+def align_width(entry_info_table)
+  keys = entry_info_table.first.keys
+  max_widths_list = get_max_width_list(keys, entry_info_table)
```

Comment:
```suggestion
  max_widths = get_max_widths(keys, entry_info_table)
```

複数形にできるならそれで配列であることを表現できるのでlistはなくても大丈夫です

Reply:
max_width_list→ max_widthsに修正しました。